### PR TITLE
Dev xxc script

### PR DESCRIPTION
**Pull request by incandescentxxc**
1. Completed DataZoom and VisualMap Option Interface and Class
2. Finished WorldPopulationTest
3. Partially Finished BeijingAQITest with some potential problems to be discussed.

### CODE REVIEW

File: src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                .setTrigger("axis")
+                .setAxisPointer(new TooltipAxisPointer()
+                        .setType("shadow"));
+        Legend legend = new Legend();
+        Grid grid = new Grid()
+                .setLeft("3%")
+                .setRight("4%")
+                .setBottom(("3%"))
+                .setContainLabel(true);
+        ValueAxis xAxis = new ValueAxis()
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:41]: Insert an empty line in between.

File: src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                .setType("value")
+                .setBoundaryGap(new Number[] {0, 0.01});
+        CategoryAxis yAxis = new CategoryAxis()
+                .setType("category")
+                .setData(new String[] {"Brazil", "Indonesia", "USA", "India", "China", "World"});
+        BarSeries series2011 = new BarSeries()
+                .setName("2011")
+                .setType("bar")
+                .setData(Arrays.asList(18203, 23489, 29034, 104970, 131744, 630230));
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:50]: We don't use List anymore. Change it to Array.

File: src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        BarSeries series2012 = new BarSeries()
+                .setName("2012")
+                .setType("bar")
+                .setData(Arrays.asList(19325, 23438, 31000, 121594, 134141, 681807));
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:54]: Use Array not List.

File: src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        Option option = new Option()
+                .setTitle(title)
+                .setTooltip(tooltip)
+                .setLegend(legend)
+                .setGrid(grid)
+                .setXAxis(xAxis)
+                .setYAxis(yAxis)
+                .setSeries(new SeriesOption[] {series2011, series2012});
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:62]: Insert a blank between "{" and "series2011".
You can use the default formatter of VSCode.

File: src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import org.icepear.echarts.component.Toolbox;
+import org.icepear.echarts.component.ToolboxDataZoomFeature;
+import org.icepear.echarts.component.ToolboxRestoreFeature;
+import org.icepear.echarts.component.ToolboxSaveAsImageFeature;
+import org.icepear.echarts.component.Tooltip;
+import org.junit.Test;
+
+public class BeijingAQITest {
+
+	private static final String path = "src/test/data/aqi-beijing.json";
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java:36]: Indent should be 4 spaces.

File: src/test/java/org/icepear/echarts/line/beijingAQITest/AQIData.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package org.icepear.echarts.line.beijingAQITest;
+
+import lombok.Data;
+
+@Data
+public class AQIData {
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/line/beijingAQITest/AQIData.java:6]: You can define this class in BeijingAQITest with "public" removed.

File: src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		Number[] data = {};
+		try {
+			data = getSeriesData();
+		} catch (Exception e) {
+			System.out.println(e);
+		}
+        // LineSeries series = new LineSeries().setName("Beijing AQI").setType("line").setData(data).setMarkLine(
+        //         new MarkLine().setSilent(true).setLineStyle(new LineStyle().setColor("#333")).setData(markLineData));
+        Option option = new Option().setDataZoom(Arrays.asList(dataZoomEle1, dataZoomEle2));
+
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java:96]: It seems you haven't finished this test.

File: src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java (hunk 2 above)
Comment (IcePear-Jzx) [reply src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:50]: Using List as a parameter is allowed, which will match with setData(Object).

Comment (IcePear-Jzx) [reply src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:50]: I think "Number[]" is better than "Object".

**Comment by IcePear-Jzx** [comment]
In general, your implementation is right, but there are some minor code style problems.

**Comment by incandescentxxc** [comment]
Some styling issues have been resolved in the last commit. Some other problems left in BeijingAQITest should be discussed before continuing.

File: src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import org.icepear.echarts.component.Tooltip;
+import org.icepear.echarts.component.TooltipAxisPointer;
+import org.icepear.echarts.component.ValueAxis;
+import org.icepear.echarts.origin.util.SeriesOption;
+import org.icepear.echarts.serializer.EChartSerializer;
+import org.junit.Test;
+
+public class WorldPopulationTest {
+
+	@Test
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/bar/WorldPopulationTest.java:27]: indent is not right

File: src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+import org.icepear.echarts.component.ToolboxSaveAsImageFeature;
+import org.icepear.echarts.component.Tooltip;
+import org.icepear.echarts.component.VisualPiece;
+import org.junit.Test;
+
+public class BeijingAQITest {
+
+	private static final String path = "src/test/data/aqi-beijing.json";
+
+	private AQIData readDTO(JsonReader reader) throws IOException {
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java:37]: indent is not right

File: src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+				.setTrigger("axis");
+
+		Grid grid = new Grid()
+				.setLeft("5%")
+				.setRight("15%")
+				.setBottom("10%");
+
+		// axis not clear
+		Map featureMap = new HashMap(); // weird
+		featureMap.put("dataZoom", new ToolboxDataZoomFeature());
```
Comment (IcePear-Jzx) [inline src/test/java/org/icepear/echarts/line/beijingAQITest/BeijingAQITest.java:79]: use HashMap() {{ put(); ... }}, so you can initialize a Map in one line.

**Comment by incandescentxxc** [comment]
Finish beijing aqi testing

**Pull request merged**